src/neuroglancer/datasource/graphene/backend.ts

```diff
@@ -0,0 +1,664 @@
+import {IndexedSegmentPropertySourceBackend} from 'neuroglancer/segmentation_display_state/backend';
+import {SkeletonChunk, SkeletonSource} from 'neuroglancer/skeleton/backend';
+import {decodeSkeletonChunk} from 'neuroglancer/skeleton/decode_precomputed_skeleton';
+import {ChunkDecoder} from 'neuroglancer/sliceview/backend_chunk_decoders';
+import {decodeCompressedSegmentationChunk} from 'neuroglancer/sliceview/backend_chunk_decoders/compressed_segmentation';
+import {decodeJpegChunk} from 'neuroglancer/sliceview/backend_chunk_decoders/jpeg';
+import {decodeRawChunk} from 'neuroglancer/sliceview/backend_chunk_decoders/raw';
+import {VolumeChunk, VolumeChunkSource} from 'neuroglancer/sliceview/volume/backend';
+import {fetchSpecialHttpByteRange} from 'neuroglancer/datasource/graphene/base';
+import {CancellationToken} from 'neuroglancer/util/cancellation';
+import {Borrowed} from 'neuroglancer/util/disposable';
+import {convertEndian32, Endianness} from 'neuroglancer/util/endian';
+import {vec3} from 'neuroglancer/util/geom';
+import {murmurHash3_x86_128Hash64Bits} from 'neuroglancer/util/hash';
+import {cancellableFetchOk, HttpError, isNotFoundError, responseArrayBuffer, responseJson} from 'neuroglancer/util/http_request';
+import {stableStringify} from 'neuroglancer/util/json';
+import {getObjectId} from 'neuroglancer/util/object_id';
+import {SpecialProtocolCredentials, SpecialProtocolCredentialsProvider} from 'neuroglancer/util/special_protocol_request';
+import {cancellableFetchSpecialOk} from 'neuroglancer/datasource/graphene/base';
+import {Uint64} from 'neuroglancer/util/uint64';
+import {encodeZIndexCompressed} from 'neuroglancer/util/zorder';
+import {registerSharedObject} from 'neuroglancer/worker_rpc';
+
+import {AnnotationSourceParameters, AnnotationSpatialIndexSourceParameters, ChunkedGraphSourceParameters, DataEncoding, GRAPHENE_MANIFEST_SHARDED, IndexedSegmentPropertySourceParameters, MeshSourceParameters, MultiscaleMeshSourceParameters, ShardingHashFunction, ShardingParameters, SkeletonSourceParameters, VolumeChunkEncoding, VolumeChunkSourceParameters} from 'neuroglancer/datasource/graphene/base';
+
+import * as DracoLoader from 'dracoloader';
+import { ChunkedGraphChunk, ChunkedGraphChunkSource, decodeSupervoxelArray } from 'src/neuroglancer/sliceview/chunked_graph/backend';
+
+
+interface ShardInfo {
+  shardUrl: string;
+  offset: Uint64;
+}
+
+interface DecodedMinishardIndex {
+  data: Uint32Array;
+  shardUrl: string;
+}
+
+interface MinishardIndexSource extends
+    GenericSharedDataSource<string, DecodedMinishardIndex|undefined> {
+  sharding: ShardingParameters;
+  credentialsProvider: SpecialProtocolCredentialsProvider;
+}
+
+function getMinishardIndexDataSource(
```

Comment:
Is this different from the precomputed sharded format?

Reply:
Hi Jeremy,

I only worked on this from the Python side, but the graphene sharded format is identical to precomputed with the exception that shard names are assigned by bounding box, not by hashing. That, and we use the regular Draco compression. This modification has only been used for meshes, I don't think there are any other differences.